lib/errgonomic.rb

```diff
@@ -88,6 +137,10 @@ def unwrap!
 
       # Return the inner value of an Ok, else raise an exception with the given
       # message when Err.
+      #
+      # @example
+      #   Ok("foo").expect!("foo") # => "foo"
+      #   Err("foo").expect!("foo") # => raise Errgonomic::ExpectError, "foo"
```

Comment:
It's unclear from this example that the original "foo" or the later "foo" is raised. Picking different values would help.

Also, rust has a lint against using foo and bar in examples for variables. One way around this is with theming. Like planet names or characters from a show you like etc.